lte/gateway/c/core/oai/tasks/amf/amf_cn.cpp

```diff
@@ -72,6 +72,27 @@ static int amf_cn_authentication_res(amf_cn_auth_res_t* const msg) {
   OAILOG_FUNC_RETURN(LOG_NAS_AMF, rc);
 }
 
+//------------------------------------------------------------------------------
+static int amf_cn_implicit_deregister_ue(const amf_ue_ngap_id_t ue_id) {
+  int rc = RETURNok;
+  struct amf_context_s* amf_ctx_p = NULL;
+
+  OAILOG_FUNC_IN(LOG_NAS_AMF);
+
+  OAILOG_DEBUG(LOG_NAS_AMF,
+               "AMF-PROC Implicit Detach UE" AMF_UE_NGAP_ID_FMT "\n", ue_id);
+
+  amf_deregistration_request_ies_t params = {};
+  params.de_reg_type = AMF_NORMAL_DEREGISTRATION;
```

Comment:
If de_reg type is AMF_NORMAL_DEREGISTRATION, then amf needs to sent de-registration accept message, but as part of implicit detach amf should not send de-registration accept message.

Reply:
Yes agreed, de-registration accept packet should not be sent out. This has been taken care in existing code. It can be seen in following pcap. There is no de registration accept sent for amf-ue-ngap-id: 1151 as it has been done implicitly.

![PR_implicit_deregistration](https://user-images.githubusercontent.com/52399057/158440565-57143a66-4517-4950-8dd6-5ba0771993d3.png)

Reply:
@Lkishor123, if de_reg type is set to AMF_NORMAL_DEREGISTRATION still AMF is not sending de-registration accept message. Could you please let us know, where is this taken care

Reply:
`amf_cn_send()` : Takes care of internal messages for core network which are not required to be sent out to gnb.